src/ovirtapi/transform.test.js

```diff
@@ -0,0 +1,343 @@
+    },
+  ],
+  expect: {
+    memory: {
+      installed:
+          {
+            firstDatum: 1073741824,
+            datum: [
+              1073741824,
+            ],
+            unit: 'bytes',
+            description: 'Total memory configured',
+          },
+      'usage.history': {
+        firstDatum: 1,
+        datum: [1, 2, 3],
+        unit: 'percent',
+        description: 'List of memory usage history, sorted by date from newest to oldest, at intervals of 30 seconds',
+      },
+      'used.history': {
+        firstDatum: 268435456,
+        datum: [ 268435456 ],
+        unit: 'bytes',
+        description: 'Memory used (agent)',
+      },
+      buffered: {
+        firstDatum: 2158592,
+        datum: [ 2158592 ],
+        unit: 'bytes',
+        description: 'Memory buffered (agent)',
+      },
+      free: {
+        firstDatum: 702443520,
+        datum: [ 702443520 ],
+        unit: 'bytes',
+        description: 'Memory free (agent)',
+      },
+    },
+    cpu: {
+      'current.total': {
+        firstDatum: 0.61,
+        datum: [0.61],
+        unit: 'percent',
+        description: 'Total CPU used',
+      },
+      'current.guest': {
+        firstDatum: 0.74,
+        datum: [0.74],
+        unit: 'percent',
+        description: 'CPU used by guest',
+      },
+      'current.hypervisor': {
+        firstDatum:
+            0.74,
+        datum: [0.74],
+        unit: 'percent',
+        description: 'CPU overhead',
+      },
+      'usage.history': {
+        firstDatum: 0,
+        datum: [0, 0, 0, 2, 0, 0],
+        unit: 'percent',
+        description: 'List of CPU usage history, sorted by date from newest to oldest, at intervals of 30 seconds',
+      },
+    },
+    network: {
+      'current.total': {
+        firstDatum: 0,
+        datum: [0],
+        unit: 'percent',
+        description: 'Total network used' },
+      'usage.history': { firstDatum: 0,
+        datum: [0, 0, 0, 0, 0, 0],
+        unit: 'percent',
+        description: 'List of CPU usage history, sorted by date from newest to oldest, at intervals of 30 seconds' } },
+    elapsedUptime: { firstDatum: 1670253,
+      datum: [1670253],
+      unit: 'seconds',
+      description: 'Elapsed VM runtime' },
+    disks: { usage: { firstDatum: { path: '/boot',
+      total: 952840192,
+      used: 100130816,
+      fs: 'ext4' },
+    datum: [{ path: '/boot',
+      total: 952840192,
+      used: 100130816,
+      fs: 'ext4' },
+    {
+      path: '/run/media/sdickers/Fedora-SB-ostree-x86_64-33',
+      total: 2874408960,
+      used: 2874408960,
+      fs: 'iso9660' }],
+    unit: 'none',
+    description: 'Disk usage, in bytes, per filesystem as JSON (agent)',
+    },
+    },
+  },
+}
+describe(testData.title, () => {
+  const result = VmStatistics.toInternal({ statistics: testData.test })
+  test(testData.title, () => {
+    expect(result).toEqual(testData.expect)
+  })
+})
```

Comment:
There is no test for a use case of an empty statistics (value is undefined) as represented here https://gerrit.ovirt.org/112146

Reply:
So just have a:
```js
  {
    title: 'undefined input',
    test: undefined,
    expect: { ... }
  }
```

and probably a few other base cases like that.

Reply:
IMO It shouldn't be tested because it's tested before the call, how the `VmStatistics.toInternal` function is implemented it's not a test case because it's tested before call: https://github.com/oVirt/ovirt-web-ui/blob/40774f083130850f7712d5fa7ce6e4affa3f2c2f/src/ovirtapi/transform.js#L216-L218
and the flow types is not include `undefined` option.
WDYT?

Reply:
IMO this shouldn't be tested since the flow types is not including undefined and in the one place we call the function we tests  this case: https://github.com/oVirt/ovirt-web-ui/blob/40774f083130850f7712d5fa7ce6e4affa3f2c2f/src/ovirtapi/transform.js#L216-L218.
If we still want to test it we should edit the function.

Reply:
I update the unit test - all the cases I could think of are now covered.